pkg/odo/cli/component/create.go

```diff
@@ -342,16 +343,18 @@ func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string
 			// Interactive mode
 			// Get component type, name and namespace from user's choice via interactive mode
 
-			// devfile.yaml is not present, user has to specify the component type
+			// devfile.yaml is not present and --devfile is not specified, user has to specify the component type
 			// Component type: We provide supported devfile component list then let you choose
-			if !util.CheckPathExists(DevfilePath) {
+			if !util.CheckPathExists(DevfilePath) && len(co.devfileMetadata.devfilePath) == 0 {
```

Comment:
We can safely compare strings in go, so `co.devfileMetadata.devfilePath == ""` should be fine (and might make the code a little easier to read)

Reply:
This is like `tab` VS `space`, firstly I am pretty sure there is no performance difference from Go compiler perspective between `len(x) == 0` and `x == ""` as Go compiler treats them the same. Seems like the discussion is pretty hot on stackoverflow as well https://stackoverflow.com/questions/18594330/what-is-the-best-way-to-test-for-an-empty-string-in-go.

I am fine for both, but the reason I choose `len(x) == 0` is consistency as in this file we have ~21 places use `len(x) == 0`

Reply:
It is about code readability, reading `o.devfileMetadata.devfilePath == ""` you understand right away what it is doing `len(co.devfileMetadata.devfilePath) == 0` now there if a function call, it it forces you to think a little bit what the author wanted from this line.

Reply:
Okay, I will make change to use `x == ""` style instead.